metric-calculation/src/metric_calculation/utils.py

```diff
@@ -104,3 +121,18 @@ def fetch_pre_etl_evidence():
 
     # Read all files into Spark.
     return SparkSession.getActiveSession().read.json(all_files)
+
+def write_metrics_to_hf_hub(
+    metrics_df: DataFrame,
+    file_output_name: str,
+    repo_id: str = "opentargets/ot-release-metrics",
+    data_dir: str = "metrics",
+    hf_token : str | None = None,
+) -> None:
+    """Upload dataset of metrics to HuggingFace Hub."""
+    assert file_output_name.endswith(".csv"), "Only CSV files are supported."
+    if hf_token:
+        login(hf_token)
+    hf_path = f"hf://datasets/{repo_id}/{data_dir}/{file_output_name}"
+    ds = Dataset.from_pandas(metrics_df.toPandas())
```

Comment:
So many conversions :p is the metrics table big?

Reply:
~100k rows. It's not terrible, but tbh most of the files are not even relevant, because we load all metrics files that were ever generated. For example, imagine for a particular release we had to generate 4 sets of metrics. We should archive the first 3 and keep only the one that reflected the data for that release.
Thanks for the review and for the suggestion.